index.js

```diff
@@ -129,6 +152,41 @@
       Object.prototype.toString.call(x).slice('[object '.length, -']'.length);
   }
 
-  return type;
+  //# parse :: String -> { namespace :: Nullable String, name :: String, version :: Number }
+  //.
+  //. Takes any String and parses it according to the [specification][3],
+  //. returning an object with a `namespace`, a `name` and a `version` field.
+  //.
+  //. ```javascript
+  //. > parse('my-package/List@2');
+  //. {namespace: 'my-package', name: 'List', version: 2}
+  //.
+  //. > parse('nonsense!');
+  //. {namespace: null, name: 'nonsense!', version: 0}
+  //.
+  //. > function Identity(x) { this.value = x; }
+  //. > Identity['@@type'] = 'my-package/Identity@1';
+  //. > parse(type(new Identity(0)));
+  //. {namespace: 'my-package', name: 'Identity', version: 1}
+  //. ```
+  function parse(s) {
+    var parsed = RPARSE.exec(s);
+    if (parsed == null) return TypeIdentifier(null, s, 0);
+    return TypeIdentifier(
+      parsed[1] == null ? null : parsed[1],
+      parsed[2],
+      parsed[3] == null ? 0 : Number(parsed[3])
+    );
+  }
+
+  return {
+    type: type,
+    parse: parse
+  };
```

Comment:
I prefer `type.parse`, though not for the reason given. I'd like to avoid `type.type`. Do you agree that `type` + `type.parse` is nicer, or are you opposed to exporting the functions in different ways?

I think the next release should be MAJOR to reflect the change to the specification, even though it is backwards compatible.